myriad-scheduler/src/main/java/org/apache/myriad/webapp/HttpConnectorProvider.java

```diff
@@ -38,11 +53,55 @@ public HttpConnectorProvider(MyriadConfiguration myriadConf) {
 
   @Override
   public Connector get() {
-    SelectChannelConnector ret = new SelectChannelConnector();
+    final AbstractConnector ret;
+    if (!myriadConf.isSSLEnabled()) {
+      final SelectChannelConnector listener = new SelectChannelConnector();
+      ret = listener;
+    } else {
+      final Configuration sslConf = new Configuration(false);
+      boolean needsClientAuth = YarnConfiguration.YARN_SSL_CLIENT_HTTPS_NEED_AUTH_DEFAULT;
+      sslConf.addResource(YarnConfiguration.YARN_SSL_SERVER_RESOURCE_DEFAULT);
+
+      final SslSocketConnectorSecure listener = new SslSocketConnectorSecure();
+      listener.setHeaderBufferSize(1024 * 64);
+      listener.setNeedClientAuth(needsClientAuth);
+      listener.setKeyPassword(getPassword(sslConf, WEB_APP_KEY_PASSWORD_KEY));
+
+      String keyStore = sslConf.get("ssl.server.keystore.location");
+
+      if (keyStore != null) {
+        listener.setKeystore(keyStore);
+        listener.setKeystoreType(sslConf.get("ssl.server.keystore.type", "jks"));
+        listener.setPassword(getPassword(sslConf, WEB_APP_KEYSTORE_PASSWORD_KEY));
```

Comment:
It is true that if password is null it will actually prompt for password, so we rely on Hadoop configuration to supply it. We could certainly give at least a warning on what may happen